### PR TITLE
chore: update arrow from v7 to v10

### LINKED ISSUE #31: Upgrade to arrow v10
arrow v8 has been released. Upgrading was included in #28 but that turns out to have causes problems when I tried to use the updated client in idpe (our internal code repo). Thus  we reverted the upgrade in #30

In order to update to arrow 8 we would need to update it in flux as well I think https://github.com/influxdata/flux 

I didn't see any open PRs to do so yet

More context is on  
https://github.com/influxdata/influxdb-iox-client-go/pull/30#issue-1235338690

### PR DESCRIPTION
**Pull request by jacobmarble**
Helps https://github.com/influxdata/conductor/issues/1114
Closes #31

This change updates Arrow from v7 to (a release candidate for) v10. Arrow v10 adds FlightSQL, which we need in a project that depends on this one.

### CODE REVIEW

**Review by domodwyer — APPROVED** [review]
Would be good to have a CI check that runs against IOx - I think we've seen breakage between versions / IOx changing versions before (I seem to remember the type of something like `COUNT(X)` changed).

File: go.mod [hunk 1]
```diff
@@ -3,29 +3,37 @@ module github.com/influxdata/influxdb-iox-client-go
 go 1.17
 
 require (
-	github.com/apache/arrow/go/v7 v7.0.0
+	github.com/apache/arrow/go/v10 v10.0.0-20220909211818-f184255cbb9b
 	github.com/influxdata/line-protocol/v2 v2.2.1
-	github.com/stretchr/testify v1.7.5
-	google.golang.org/grpc v1.44.0
+	github.com/stretchr/testify v1.8.0
+	google.golang.org/grpc v1.49.0
```
Comment (onelson) [inline go.mod:9]: iirc last time we bumped grpc along with arrow, there had to be some alignment in C2 etc. It took a bit of a bit of a team effort to vet.

Is there a similar concern this time? Maybe all affected are already on board, or the grpc bump is less impactful this time? :shrug:

Comment (mkmik) [reply go.mod:9]: We probably should bump grpc in a separate PR in idpe before upgrading its dependency to arrow

Comment (mkmik) [reply go.mod:9]: Also, dependabot and friends are a good idea also to prevent situations like this

Comment (mkmik) [reply go.mod:9]: Here https://github.com/influxdata/idpe/pull/15779

**Comment by jacobmarble** [comment]
@onelson any hesitation about a pre-release version of Arrow? If so, there's probably a way to import two versions of one module in IDPE, I just need to google it.

**Comment by onelson** [comment]
@jacobmarble I floated the question to the team but nobody seemed concerned?

I wondered if we'd need to align our arrow in Flux - if it would be required, or even if there'd be a perf penalty if we didn't... I'd like to check to see what breaks in flux (if anything) if we try to align.

**Comment by onelson** [comment]
@jacobmarble alright - after conferring with the team today, here's where we are on the Flux side.

We're happy to see this upgrade with the following provisions:

- the iox client bumps to a _new major version_ such that Flux can stay on the previous until we're ready to upgrade arrow to v10 there
- the v7-based client will continue to work for us in the mean time

We will need to set aside some time to look at the arrow usage in the flux internals before we even know if there are challenges.

As far as versioning goes, my understanding is multiple versions of a Go package can be used side-by-side so long as they are each different major releases and it seems like since the types from arrow are a public part of the client API, bumping arrow should necessarily mean a major bump for this client.

Does that track with y'all?

**Comment by jacobmarble** [comment]
@onelson this plan works for me. This isn't a versioned golang module yet, so I'll work on that.

**Pull request merged**